core/src/main/java/com/rudderstack/android/sdk/core/DBPersistentManager.java

```diff
@@ -37,6 +39,16 @@ private void createSchema(SQLiteDatabase db) {
      * save individual messages to DB
      * */
     void saveEvent(String messageJson) {
+        try {
+            if (messageJson.getBytes("UTF-8").length > Utils.MAX_EVENT_SIZE) {
+                RudderLogger.logError(String.format(Locale.US, "DBPersistentManager: saveEvent: Event size exceeds the maximum permitted event size(%d)",
```

Comment:
Keep this logic to EventRepository as this file should be responsible for handling database stuff

Reply:
done